File: javascript/remove_duplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if(nums.length ==0){
        return 0;
    }

    let slow = 0;
    for (let fast = 1; fast < nums.length; fast++){
        if(nums[fast] != nums[slow]){
            slow ++;

          nums[slow]= nums[fast];


        }
    }
    return slow + 1;
};

//example usage
let arr = [1, 1, 2, 2, 3, 8, 9];
let newLength = removeDuplicates(arr);
console.log(newLength); // Output: 5
console.log(arr.slice(0, newLength)); // Output: [1, 2, 3, 8, 9]

File: dart/remove_duplicates.dart
class Solution {
  int removeDuplicates(List<int> nums) {
    if (nums.isEmpty) {
      return 0;
    }
    int slow = 0;
    for (int fast = 1; fast < nums.length; fast++) {
      if (nums[fast] != nums[slow]) {
        slow++;
        nums[slow] = nums[fast];
      }
    }
    return slow + 1;
  }
}
